src/fireedge/src/modules/components/Forms/Marketplace/CreateForm/Steps/Configuration/fieldsS3.js

```diff
@@ -0,0 +1,207 @@
+ * Unless required by applicable law or agreed to in writing, software       *
+ * distributed under the License is distributed on an "AS IS" BASIS,         *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+ * See the License for the specific language governing permissions and       *
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+import { INPUT_TYPES, T, MARKET_TYPES } from '@ConstantsModule'
+import { string, boolean, number } from 'yup'
+import { Field } from '@UtilsModule'
+
+/** @type {Field} AWS field */
+const AWS = {
+  name: 'AWS',
+  label: T['marketplace.form.configuration.s3.aws'],
+  tooltip: T['marketplace.form.configuration.s3.aws.tooltip'],
+  type: INPUT_TYPES.SWITCH,
+  dependOf: '$general.MARKET_MAD',
+  htmlType: (type) => type !== MARKET_TYPES.S3.value && INPUT_TYPES.HIDDEN,
+  validation: boolean()
+    .yesOrNo()
+    .afterSubmit((value, { context }) => {
+      if (context?.general?.MARKET_MAD === MARKET_TYPES.S3.value) {
+        return value ? 'YES' : 'NO'
+      } else {
+        return undefined
+      }
+    })
+    .default(() => true),
+  grid: { xs: 12, md: 6 },
+}
+
+/** @type {Field} ACCESS_KEY_ID field */
+const ACCESS_KEY_ID = {
+  name: 'ACCESS_KEY_ID',
+  label: T['marketplace.form.configuration.s3.accessKey'],
+  tooltip: T['marketplace.form.configuration.s3.accessKey.tooltip'],
+  type: INPUT_TYPES.TEXT,
+  dependOf: '$general.MARKET_MAD',
+  htmlType: (type) => type !== MARKET_TYPES.S3.value && INPUT_TYPES.HIDDEN,
+  validation: string()
+    .trim()
+    .when(
+      '$general.MARKET_MAD',
+      (type, schema) =>
+        type &&
+        (type !== MARKET_TYPES.S3.value ? schema.strip() : schema.required())
+    )
+    .default(() => undefined),
+  grid: { md: 12 },
+}
+
+/** @type {Field} SECRET_ACCESS_KEY field */
+const SECRET_ACCESS_KEY = {
+  name: 'SECRET_ACCESS_KEY',
+  label: T['marketplace.form.configuration.s3.secretAccessKey'],
+  tooltip: T['marketplace.form.configuration.s3.secretAccessKey.tooltip'],
+  type: INPUT_TYPES.TEXT,
+  dependOf: '$general.MARKET_MAD',
+  htmlType: (type) => type !== MARKET_TYPES.S3.value && INPUT_TYPES.HIDDEN,
+  validation: string()
+    .trim()
+    .when(
+      '$general.MARKET_MAD',
+      (type, schema) =>
+        type &&
+        (type !== MARKET_TYPES.S3.value ? schema.strip() : schema.required())
+    )
+    .default(() => undefined),
+  grid: { md: 12 },
+}
+
+/** @type {Field} BUCKET field */
+const BUCKET = {
+  name: 'BUCKET',
+  label: T['marketplace.form.configuration.s3.bucket'],
+  tooltip: T['marketplace.form.configuration.s3.bucket.tooltip'],
+  type: INPUT_TYPES.TEXT,
+  dependOf: '$general.MARKET_MAD',
+  htmlType: (type) => type !== MARKET_TYPES.S3.value && INPUT_TYPES.HIDDEN,
+  validation: string()
+    .trim()
+    .when(
+      '$general.MARKET_MAD',
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).